Gems/PhysX/Code/Tests/CharacterGameplayControllerTests.cpp

```diff
@@ -0,0 +1,127 @@
+            AzPhysics::SceneHandle sceneHandle,
+            float gravityMultiplier = DefaultGravityMultiplier,
+            float groundDetectionBoxHeight = DefaultGroundDetectionBoxHeight,
+            const AZ::Transform& floorTransform = DefaultFloorTransform)
+            : m_sceneHandle(sceneHandle)
+        {
+            if (auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get())
+            {
+                m_testScene = physicsSystem->GetScene(m_sceneHandle);
+            }
+            AZ_Assert(m_testScene != nullptr, "GameplayTestBasis: the Test scene is null.");
+            SetUp(gravityMultiplier, groundDetectionBoxHeight, floorTransform);
+        }
+
+        void SetUp(float gravityMultiplier = DefaultGravityMultiplier,
+            float groundDetectionBoxHeight = DefaultGroundDetectionBoxHeight,
+            const AZ::Transform& floorTransform = DefaultFloorTransform)
+        {
+            m_floor = PhysX::TestUtils::AddStaticFloorToScene(m_sceneHandle, floorTransform);
+
+            m_controllerEntity = AZStd::make_unique<AZ::Entity>("CharacterEntity");
+
+            // Transform Setup
+            m_controllerEntity->CreateComponent<AzFramework::TransformComponent>()->SetWorldTM(AZ::Transform::Identity());
+
+            // Character Controller Setup
+            auto capsuleShapeConfiguration = AZStd::make_unique<Physics::CapsuleShapeConfiguration>();
+            auto characterConfiguration = AZStd::make_unique<Physics::CharacterConfiguration>();
+            characterConfiguration->m_maximumSlopeAngle = 25.0f;
+            characterConfiguration->m_stepHeight = 0.2f;
+
+            m_controllerEntity->CreateComponent<CharacterControllerComponent>(
+                AZStd::move(characterConfiguration), AZStd::move(capsuleShapeConfiguration));
+
+            // Gameplay Controller Setup
+            PhysX::CharacterGameplayConfiguration characterGameplayConfiguration;
+
+            characterGameplayConfiguration.m_gravityMultiplier = gravityMultiplier;
+            characterGameplayConfiguration.m_groundDetectionBoxHeight = groundDetectionBoxHeight;
+            m_gameplayController = m_controllerEntity->CreateComponent<CharacterGameplayComponent>(characterGameplayConfiguration);
+            
+            // Wrap UP
+            m_controllerEntity->Init();
+            m_controllerEntity->Activate();
+
+            Physics::CharacterRequestBus::EventResult(m_controller, m_controllerEntity->GetId(), &Physics::CharacterRequests::GetCharacter);
+        }
+
+        void Update(AZ::u32 numTimeSteps = 1)
+        {
+            if (auto* physXSystem = GetPhysXSystem())
+            {
+                for (AZ::u32 i = 0; i < numTimeSteps; i++)
+                {
+                    physXSystem->Simulate(m_timeStep);
+                }
+            }
+        }
+
+        AzPhysics::Scene* m_testScene;
+        AzPhysics::SceneHandle m_sceneHandle;
+        AzPhysics::StaticRigidBody* m_floor;
```

Comment:
nit: please initialize these to nullptr as well.

````
        AzPhysics::Scene* m_testScene = nullptr;
        AzPhysics::SceneHandle m_sceneHandle = nullptr;
        AzPhysics::StaticRigidBody* m_floor = nullptr;
````

Reply:
Scenehandle would not accept a `nullptr`. Added the other two.